src/components/connect.js

```diff
@@ -279,9 +280,25 @@ function connect(mapPropsToRequestsToProps, defaults) {
       }
 
       componentWillReceiveProps(nextProps, nextContext) {
+        if (defaults.pure) {
+          this._propsAndContextDidntChange = shallowEqual(this.props, nextProps) &&
```

Comment:
if anyone can think of a better name for `this._propsAndContextDidntChange` I'm all ears :)

(I save it to a variable to avoid recomputing it `shouldComponentUpdate` (which, if props or context changed, always runs after componentWillReceiveProps))